alacritty_terminal/src/config/mod.rs

```diff
@@ -418,17 +401,50 @@ impl Default for DefaultTrueBool {
     }
 }
 
+fn defaulted<T, E>(err: E) -> T
+    where T: Default, E: Display
+{
+    error!("Problem with config: {}; using default value", err);
+    T::default()
+}
+
 pub fn failure_default<'a, D, T>(deserializer: D) -> Result<T, D::Error>
 where
     D: Deserializer<'a>,
     T: Deserialize<'a> + Default,
 {
-    let value = serde_yaml::Value::deserialize(deserializer)?;
-    match T::deserialize(value) {
-        Ok(value) => Ok(value),
-        Err(err) => {
-            error!("Problem with config: {}; using default value", err);
-            Ok(T::default())
-        },
-    }
+    Ok(T::deserialize(Value::deserialize(deserializer)?).unwrap_or_else(defaulted))
+}
+
+pub fn none_str_as_none<'de, T, D>(deserializer: D) -> Result<Option<T>, D::Error>
+    where D: Deserializer<'de>, T: Deserialize<'de> + Default
+{
+    Ok(match Value::deserialize(deserializer)? {
+        Value::String(ref value) if value.to_lowercase() == "none" => None,
+        value => Some(T::deserialize(value).unwrap_or_else(defaulted))
+    })
+}
+
+pub fn abbrev_from_str<'de, T, D>(deserializer: D) -> Result<T, D::Error>
+where D: Deserializer<'de>, T: Deserialize<'de> + From<String> + Default
+{
+    Ok(match Value::deserialize(deserializer)? {
+        Value::String(value) => T::from(value),
+        value => T::deserialize(value).unwrap_or_else(defaulted)
+    })
+}
+
+pub fn option_abbrev_from_str<'de, T, D>(deserializer: D) -> Result<Option<T>, D::Error>
+where D: Deserializer<'de>, T: Deserialize<'de> + From<String>
+{
+    Ok(match Value::deserialize(deserializer)? {
+        Value::String(value) => Some(T::from(value)),
+        value => match T::deserialize(value) {
+            Ok(value) => Some(value),
+            Err(err) => {
+                error!("Problem with config: {}; using None", err);
+                None
+            }
+        }
+    })
 }
```

Comment:
Rust's trait coherence rules do not allow it since both `From` and `Option` are not defined in this crate.

Reply:
Yeah that's true, but we could still just write a thin wrapper around Option<Shell> and that would allow us to implement it.

I still think that just implementing Default for shell would be the easiest solution here, but allowing all fields to be optional might be an alternative to write a completely separate type wrapping around Option<Shell>.

Reply:
I'm inclined to leave it like this since both functions are generically useful.  Future settings may not have a meaningful `Default` but can still benefit from an shorthand string form.

Reply:
There's another option, which is creating a custom `FromString` trait. This allows us to implement it for any type, including wrappers from std like `Arc` or `Option`.

Since this works very well, I've decided to go for this and have made the required changes myself. I think that's a good solution for the problem and it removes the need for two functions which are extremely similar.